lib/msf/core/auxiliary/vyos.rb

```diff
@@ -0,0 +1,304 @@
+# -*- coding: binary -*-
+
+require 'metasploit/framework/hashes/identify'
+
+module Msf
+  ###
+  #
+  # This module provides methods for working with VyOS equipment
+  #
+  ###
+  module Auxiliary::VYOS
+    include Msf::Auxiliary::Report
+
+    def vyos_config_eater(thost, tport, config, store = true)
+
+      credential_data = {
+        address: thost,
+        port: tport,
+        protocol: 'tcp',
+        workspace_id: myworkspace.id,
+        origin_type: :service,
+        private_type: :nonreplayable_hash,
+        # jtr_format: 'sha512,crypt', # default on the devices 11.4.0+
+        service_name: '',
+        module_fullname: fullname,
+        status: Metasploit::Model::Login::Status::UNTRIED
+      }
+
+      # Default SNMP to UDP
+      if tport == 161
+        credential_data[:protocol] = 'udp'
+      end
+
+      if store && !config.include?('such file or directory')
```

Comment:
I also got `"cat: /config/config.boot: Permission denied"` error with a non-privileged user. Maybe, this should also be handled to avoid storing this string in the loot.

Also, I think checking if the `config` string contains actual data would be better located in the module, before calling `vyos_config_eater `.

Reply:
doesn't matter to me either way, however if it were located in the module, it would need to be replicated to the aux/admin module as well.  I figured one place would make it more fool proof